x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -185,6 +195,45 @@ public static long chronoToLong(long dateTime, ChronoField chronoField, ZoneId z
         return Instant.ofEpochMilli(dateTime).atZone(zone).getLong(chronoField);
     }
 
+    /**
+     * The following conversions are between String and other data types.
+     */
+    public static BytesRef stringToIP(BytesRef field) {
```

Comment:
> Nit: s/field/value (here and below), or something else, as it doesn't have to be a field, can be any value. But if we want these to be used mostly for cases where the input is field value, we might want to log a comment.

I'll try to give them more consistent and clear names. If it is a primitive/number, I'll call it number. If it is an instance of a class, I'll call it field. If there is something obviously better and meaningful, I'll use it in stead of a generic name.

Reply:
> Some of the functions here are just a renaming wrapper for functionality either in some utils or JDK, which I can see bring uniformity to conversion primitives in ESQL, but might seem like a bit of confusing duplication. So I was wondering if we don't want to keep this class focused on the converters [infrastructure] and move these utils somewhere (maybe QL, like to `StringUtils` and `NumericUtils` or some ESQL-specific ones).
> 
> Anyways, just mentioning, this can also stay as is.

Yeah, I was thinking about the same thing, it is a bit of confusing duplication for the conversions between primitives and those just call StringUtils and NumericUtils. As we are trying to make EsqlDataTypeConverter a central place for data type conversions within ESQL, I put all(or an anchor of those) of the conversions that I can find so far here. I was thinking of make a copy of the utility methods/functions used in QL into EsqlDataTypeConverter, but I haven't done it yet. There is a bigger effort of grand refactor #106679, perhaps it can be a subtask of it, so that it can be done in a more consistent way.